syntax = "proto3";

option java_multiple_files = true;
option java_package = "gRPC.com.protos";
option java_outer_classname = "UserProto";

service UserService {
    rpc GetAllUsers(Empty) returns (UsersList) {}
    rpc GetUser(Id) returns (User) {}
    rpc CreateUser(UserDTO) returns (Empty) {}
    rpc AddData(stream DataRequest) returns (Empty) {}
    rpc GetData(Id) returns (stream DataResponse) {}
    rpc ExchangeData(stream DataRequest) returns (stream DataResponse) {}
}

message Empty {

}

message User {
    int32 id = 1;
    string name = 2;
    bytes data = 3;
}

message UserDTO {
    string name = 1;
}

message UsersList {
    repeated User users = 1;
}

message Id {
    int32 id = 1;
}

message DataRequest {
    int32 id = 1;
    bytes data = 2;
}

message DataResponse {
    bytes data = 1;
}
